src/generators/default_map_generator_job.cpp

```diff
@@ -784,28 +764,42 @@ std::string default_map_generator_job::default_generate_map(size_t width, size_t
 		} else {
 			base_name_generator = base_generator_factory.get_name_generator("male");
 		}
-		river_name_generator	= base_generator_factory.get_name_generator("river");
-		lake_name_generator		= base_generator_factory.get_name_generator("lake");
-		road_name_generator		= base_generator_factory.get_name_generator("road");
+
+		river_name_generator    = base_generator_factory.get_name_generator("river");
+		lake_name_generator     = base_generator_factory.get_name_generator("lake");
+		road_name_generator     = base_generator_factory.get_name_generator("road");
 		bridge_name_generator   = base_generator_factory.get_name_generator("bridge");
 		mountain_name_generator = base_generator_factory.get_name_generator("mountain");
 		forest_name_generator   = base_generator_factory.get_name_generator("forest");
-		swamp_name_generator	= base_generator_factory.get_name_generator("swamp");
+		swamp_name_generator    = base_generator_factory.get_name_generator("swamp");
 	}
 
-	std::vector<terrain_height_mapper> height_conversion;
+	// Generate the height of everything.
+	const height_map heights = generate_height_map(width, height, iterations, hill_size, island_size, island_off_center);
+
+	LOG_NG << "Done generating height map. " << (SDL_GetTicks() - ticks) << " ticks elspased" << "\n";
+	ticks = SDL_GetTicks();
 
-	for(const config &h : cfg.child_range("height")) {
+	// Find out what the 'flatland' on this map is, i.e. grassland.
```

Comment:
Oh, here's that code from above. Hard to tell if it changed or is a straight copy-paste, though.